packages/x-components/src/x-modules/extra-params/components/__tests__/renderless-extra-params.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { resetXExtraParamStateWith } from './utils';
+
+describe('testing RenderlessExtraParam component', () => {
+  function renderRenderlessExtraParams({
+    scopedSlots,
+    defaultValue,
+    extraParamName,
+    params
+  }: RenderlessExtraParamsOptions): RenderlessExtraParamsAPI {
+    const localVue = createLocalVue();
+    localVue.use(Vuex);
+
+    const store: Store<DeepPartial<RootXStoreState>> = new Store({});
+    installNewXPlugin({ store }, localVue);
+
+    resetXExtraParamStateWith(store, {
+      params: params ?? {}
+    });
+
+    const wrapper = mount(RenderlessExtraParam, {
+      propsData: {
+        extraParamName,
+        defaultValue
+      },
+      localVue,
+      store,
+      scopedSlots
+    });
+
+    return {
+      wrapper
+    };
+  }
+
+  it('is an XComponent which has an XModule', () => {
+    const { wrapper } = renderRenderlessExtraParams({ extraParamName: 'warehouse' });
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('extraParams');
+  });
+
+  it('renders a custom slot content', () => {
+    const { wrapper } = renderRenderlessExtraParams({
+      extraParamName: 'warehouse',
+      scopedSlots: { default: `<span data-test="custom-slot">Custom slot</span>` }
+    });
+
+    expect(wrapper.find(getDataTestSelector('custom-slot')).text()).toEqual('Custom slot');
+  });
+
+  // eslint-disable-next-line max-len
+  it("emits ExtraParamsProvided event when the component receives a default value and and it doesn't exist in the store", () => {
+    const extraParamsProvidedCallback = jest.fn();
+    const { wrapper } = renderRenderlessExtraParams({
+      extraParamName: 'warehouse',
+      defaultValue: 1234
+    });
+
+    wrapper.vm.$x.on('ExtraParamsProvided', true).subscribe(extraParamsProvidedCallback);
+
+    expect(extraParamsProvidedCallback).toHaveBeenNthCalledWith<[WirePayload<Dictionary<unknown>>]>(
+      1,
+      {
+        eventPayload: { warehouse: 1234 },
+        metadata: { moduleName: 'extraParams' }
+      }
+    );
+  });
+
+  // eslint-disable-next-line max-len
+  it("not emits ExtraParamsProvided event when the component receives a default value if it's in the store", () => {
+    const extraParamsProvidedCallback = jest.fn();
+
+    const { wrapper } = renderRenderlessExtraParams({
+      extraParamName: 'warehouse',
+      defaultValue: 1234,
+      params: { warehouse: 1234 }
+    });
+
+    wrapper.vm.$x.on('ExtraParamsProvided', true).subscribe(extraParamsProvidedCallback);
+
+    expect(extraParamsProvidedCallback).toHaveBeenCalledTimes(0);
+  });
+
+  it('emits UserChangedExtraParams event when the update method is called', () => {
+    const userChangedExtraParamsCallback = jest.fn();
+    const { wrapper } = renderRenderlessExtraParams({
+      extraParamName: 'warehouse',
+      scopedSlots: {
+        default: `
+          <template #search="{ defaultValue, updateValue }">
```

Comment:
I don't know how this is working, you are defining a scoped slot that has a scoped slot named `search` 🤣,  For me it is far more clear if we just use the template like in other tests. Here I have exactly the input of the test, properly defined, a template with a default scoped slot.

```ts
    const { wrapper } = renderRenderlessExtraParams({
      extraParamName: 'warehouse',
      template: `
        <RenderlessExtraParam #default="{ defaultValue, updateValue }">
          <button data-test="custom-slot" @click="updateValue(45678)">Update warehouse</button>
        </RenderlessExtraParam>`
    });

```

 Moreover, wasn't that scoped slots property giving us some troubles?

```
    const { wrapper } = renderRenderlessExtraParams({
      extraParamName: 'warehouse',
      template: `
        <RenderlessExtraParam #default="{ defaultValue, updateValue }">
          <button data-test="custom-slot" @click="updateValue(45678)">Update warehouse</button>
        </RenderlessExtraParam>`
    });
```